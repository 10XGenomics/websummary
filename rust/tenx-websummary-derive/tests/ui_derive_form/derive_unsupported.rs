use tenx_websummary_derive::HtmlForm;

#[derive(HtmlForm)]
enum Foo {
    Bar {},
    Bar2,
}

#[derive(HtmlForm)]
enum Foo2 {
    Bar(u8),
    Bar2,
}

#[derive(HtmlForm)]
struct Boo(u8);

fn main() {}
