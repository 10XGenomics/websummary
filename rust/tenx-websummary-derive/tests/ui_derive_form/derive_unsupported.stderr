error: HtmlForm can only be derived for structs with named fields or enum with unit variants. Unsupported shape `named`
 --> tests/ui_derive_form/derive_unsupported.rs:3:10
  |
3 | #[derive(HtmlForm)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `HtmlForm` (in Nightly builds, run with -Z macro-backtrace for more info)

error: HtmlForm can only be derived for structs with named fields or enum with unit variants. Unsupported shape `newtype`
 --> tests/ui_derive_form/derive_unsupported.rs:9:10
  |
9 | #[derive(HtmlForm)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `HtmlForm` (in Nightly builds, run with -Z macro-backtrace for more info)

error: HtmlForm can only be derived for structs with named fields or enum with unit variants. Unsupported shape `newtype`
  --> tests/ui_derive_form/derive_unsupported.rs:15:10
   |
15 | #[derive(HtmlForm)]
   |          ^^^^^^^^
   |
   = note: this error originates in the derive macro `HtmlForm` (in Nightly builds, run with -Z macro-backtrace for more info)
